File: 05_Open_source_ProjectRepository/AIAgentFramework/LuminaRuntimeContainer/examples/cpp/hello.cpp
// Lumina Runtime Container - C++示例

#include <iostream>
#include <string>
#include <vector>

// 主函数
int main() {
    // 打印欢迎信息
    std::cout << "====================================" << std::endl;
    std::cout << "Hello from Lumina Runtime Container!" << std::endl;
    std::cout << "====================================" << std::endl;
    std::cout << "Language: C++" << std::endl;
    std::cout << "Container Version: 1.0.0" << std::endl;
    std::cout << "====================================" << std::endl;
    
    // 演示基本功能
    std::cout << "\nBasic C++ Features:" << std::endl;
    
    // 变量和输出
    int number = 42;
    std::string message = "The answer to life, the universe, and everything";
    std::cout << "Integer variable: " << number << std::endl;
    std::cout << "String variable: " << message << std::endl;
    
    // 循环
    std::cout << "\nLoop demonstration:" << std::endl;
    for (int i = 1; i <= 5; ++i) {
        std::cout << "Iteration " << i << std::endl;
    }
    
    // 向量
    std::cout << "\nVector demonstration:" << std::endl;
    std::vector<std::string> languages = {"C++", "Python", "Java", "JavaScript", "Go"};
    for (const auto& lang : languages) {
        std::cout << "Supported language: " << lang << std::endl;
    }
    
    // 函数调用
    std::cout << "\nFunction demonstration:" << std::endl;
    
    // 计算平方的函数
    auto square = [](int x) {
        return x * x;
    };
    
    for (int i = 1; i <= 3; ++i) {
        std::cout << "Square of " << i << " is " << square(i) << std::endl;
    }
    
    // 结束信息
    std::cout << "====================================" << std::endl;
    std::cout << "C++ example completed successfully!" << std::endl;
    std::cout << "====================================" << std::endl;
    
    return 0;
}

File: 05_Open_source_ProjectRepository/AIAgentFramework/LuminaRuntimeContainer/examples/python/hello.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
Lumina Runtime Container - Python示例
"""

import os
import sys
import time

# 打印欢迎信息
print("====================================")
print("Hello from Lumina Runtime Container!")
print("====================================")
print(f"Language: Python {sys.version}")
print(f"Container Version: 1.0.0")
print(f"Current directory: {os.getcwd()}")
print("====================================")

# 演示基本功能
print("\nBasic Python Features:")

# 变量和输出
number = 42
message = "The answer to life, the universe, and everything"
print(f"Integer variable: {number}")
print(f"String variable: {message}")

# 循环
print("\nLoop demonstration:")
for i in range(1, 6):
    print(f"Iteration {i}")

# 列表
print("\nList demonstration:")
languages = ["C++", "Python", "Java", "JavaScript", "Go"]
for lang in languages:
    print(f"Supported language: {lang}")

# 函数
print("\nFunction demonstration:")

def square(x):
    """计算平方"""
    return x * x

for i in range(1, 4):
    print(f"Square of {i} is {square(i)}")

# 尝试导入AI框架
print("\nAI Framework Demonstration:")

try:
    import tensorflow as tf
    print(f"TensorFlow version: {tf.__version__}")
    # 简单的TensorFlow示例
    a = tf.constant([[1.0, 2.0], [3.0, 4.0]])
    b = tf.constant([[5.0, 6.0], [7.0, 8.0]])
    c = tf.matmul(a, b)
    print(f"TensorFlow matrix multiplication result:\n{c.numpy()}")
except ImportError:
    print("TensorFlow not installed")

try:
    import torch
    print(f"PyTorch version: {torch.__version__}")
    # 简单的PyTorch示例
    x = torch.tensor([[1.0, 2.0], [3.0, 4.0]])
    y = torch.tensor([[5.0, 6.0], [7.0, 8.0]])
    z = torch.matmul(x, y)
    print(f"PyTorch matrix multiplication result:\n{z.numpy()}")
except ImportError:
    print("PyTorch not installed")

try:
    import numpy as np
    print(f"NumPy version: {np.__version__}")
    # 简单的NumPy示例
    arr = np.array([1, 2, 3, 4, 5])
    print(f"NumPy array: {arr}")
    print(f"NumPy array mean: {np.mean(arr)}")
except ImportError:
    print("NumPy not installed")

# 结束信息
print("\n====================================")
print("Python example completed successfully!")
print("====================================")
